src/Assets/js/tasks.js

```diff
@@ -0,0 +1,20 @@
+function runTask(id) {
+    var btn = document.getElementById('runBtn' + id);
+
+    var xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function () {
+        if (this.readyState < 4) {
+            btn.setAttribute('onclick', 'alert("Please wait!")');
```

Comment:
Instead of repeatedly updating onclick, why not do this?

 - Set the button to be disabled when it is first clicked
 - Add a button disabled check to this func (can still have a fallback to the alert if the user manages to click a disabled button)
 - Un-disable the button once the request completes